src/diff/index.js

```diff
@@ -270,9 +270,9 @@ function diffElementNodes(dom, newVNode, oldVNode, context, isSvg, excessDomChil
 					dom.innerHTML = newHtml && newHtml.__html || '';
 				}
 			}
-			if (newProps.multiple) {
-				dom.multiple = newProps.multiple;
-			}
+			// Needs to be set before diffing children for <select>'s with multiple
+			// options.
+			if (newProps.multiple) dom.multiple = newProps.multiple;
```

Comment:
Is this still needed with children now being diffed before props?

Reply:
Unfortunately not. This issue is caused when the value is set before the range boundaries are applied. At the time when value is set, the browser will default to the fallback range of 0-100 and won't update `value` when `min/max/step` are added later.

Reply:
yeah the multiple property must be set before `children` are diffed, otherwise there will be only one `<option>` with a selected value. That's the only exception we've found so far :/